Jenkinsfile

```diff
@@ -1,42 +1,107 @@
 library 'magic-butler-catalogue'
 def PROJECT_NAME = 'logdna-agent-v2'
 
+def boolean changesMatch(String pathRegex) {
+    return !env.CHANGE_TARGET || sh(
+        script: "git diff --name-only origin/${env.CHANGE_TARGET}...${env.GIT_COMMIT} | grep \"${pathRegex}\"",
+        returnStatus: true
+    ) == 0
+}
+
+def RUST_IMAGE = 'docker.pkg.github.com/answerbook/docker-images/logdna-agent-rust:latest'
+
 pipeline {
     agent any
-
     options {
         timestamps()
         ansiColor 'xterm'
     }
     stages {
-        stage('Build') {
-            parallel {
-                stage('Build Test Dependencies') {
-                    steps {
-                        sh 'make test-deps'
+        stage('Build Rust Image') {
+            when {
+                anyOf {
+                    expression {
+                        return changesMatch('^Makefile$')
                     }
-                }
-                stage('Build Agent') {
-                    steps {
-                        sh 'make build'
+                    expression {
+                        return changesMatch('^rust-image/')
```

Comment:
What are you trying to accomplish here?

Reply:
I explained the base image build process in a different comment, but this is to detect changes for the files used to build the rust base image. We only want to build a local version if it's changed. Additionally we want to see the entire difference for the entire branch/PR hence the use of `changesMatch` instead of Jenkins `changeSet`